File: backend/src/main/java/com/errorquest/entity/ErrorQuestion.java
package com.errorquest.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("error_question")
public class ErrorQuestion {
    
    @TableId(type = IdType.AUTO)
    private Long questionId;
    
    private Long userId;
    
    private Long subjectId;
    
    private Long knowledgeId;
    
    private Integer questionType;
    
    private Integer difficulty;
    
    private String errorReason;
    
    private String questionContent;
    
    private String questionImages;
    
    private String options;
    
    private String correctAnswer;
    
    private String answerImages;
    
    private String userAnswer;
    
    private String userAnswerImages;
    
    private String analysis;
    
    private String notes;
    
    private String tags;
    
    private Integer masteryStatus;
    
    @TableLogic
    private Integer deleted;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: frontend/src/api/error-question.ts
import request from '@/utils/request'

export interface ErrorQuestion {
  questionId?: number
  subjectId: number
  knowledgeId?: number
  questionType: number
  difficulty: number
  errorReason: string
  questionContent: string
  questionImages?: string
  options?: string
  correctAnswer: string
  answerImages?: string
  userAnswer: string
  userAnswerImages?: string
  analysis?: string
  notes?: string
  tags?: string
  masteryStatus?: number
  createTime?: string
}

export interface ErrorQuestionQuery {
  subjectId?: number
  knowledgeId?: number
  questionType?: number
  difficulty?: number
  errorReason?: string
  keyword?: string
  masteryStatus?: number
  pageNum?: number
  pageSize?: number
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export function addErrorQuestion(data: ErrorQuestion): Promise<void> {
  return request.post('/error-questions', data)
}

export function updateErrorQuestion(id: number, data: ErrorQuestion): Promise<void> {
  return request.put(`/error-questions/${id}`, data)
}

export function deleteErrorQuestion(id: number): Promise<void> {
  return request.delete(`/error-questions/${id}`)
}

export function getErrorQuestionDetail(id: number): Promise<ErrorQuestion> {
  return request.get(`/error-questions/${id}`)
}

export function listErrorQuestions(params: ErrorQuestionQuery): Promise<PageResult<ErrorQuestion>> {
  return request.get('/error-questions', { params })
}

export function updateMasteryStatus(id: number, status: number): Promise<void> {
  return request.put(`/error-questions/${id}/mastery`, null, { params: { status } })
}
